CRM/Civicase/Page/ContactActivityTab.php

```diff
@@ -7,4 +7,39 @@
  */
 class CRM_Civicase_Page_ContactActivityTab extends CRM_Core_Page {
 
+  /**
+   * Run function of the class.
+   */
+  public function run() {
+    $this->preProcess();
+
+    return parent::run();
+  }
+
+  /**
+   * Pre process function.
+   */
+  public function preProcess() {
+    $this->_contactId = CRM_Utils_Request::retrieve('cid', 'Positive', $this, TRUE);
+    $this->assign('contactId', $this->_contactId);
+    // FIXME: need to fix this conflict.
+    $this->assign('contactID', $this->_contactId);
+
+    // Check logged in url permission.
+    CRM_Contact_Page_View::checkUserPermission($this);
+
+    $this->_action = CRM_Utils_Request::retrieve('action', 'String', $this, FALSE, 'browse');
+    $this->assign('action', $this->_action);
+
+    // Also create the form element for the activity links box.
+    $controller = new CRM_Core_Controller_Simple(
+      'CRM_Activity_Form_ActivityLinks',
+      ts('Activity Links'),
+      NULL,
+      FALSE, FALSE, TRUE
+    );
+    $controller->setEmbedded(TRUE);
+    $controller->run();
```

Comment:
Seems the only part of the code we need is 

```
   $controller = new CRM_Core_Controller_Simple(
      'CRM_Activity_Form_ActivityLinks',
      ts('Activity Links'),
      NULL,
      FALSE, FALSE, TRUE
    );
    $controller->setEmbedded(TRUE);
    $controller->run();
```
This is the part that instantiates the activity links form.

Can you confirm if this works?

Reply:
For the `$action` and `$permission` to have values, we need the other lines.

Reply:
I see. Can we try clean up these function a bit then so we have just what we need to make this work? The lesser code we copy to get this working, the better. Can you try this ? I removed some part I think is not needed

```
  public function preProcess() {
    $this->_contactId = CRM_Utils_Request::retrieve('cid', 'Positive', $this, TRUE);

    // check logged in url permission
    CRM_Contact_Page_View::checkUserPermission($this);

    $this->_action = CRM_Utils_Request::retrieve('action', 'String', $this, FALSE, 'browse');
    $this->assign('action', $this->_action);

    // also create the form element for the activity links box
    $controller = new CRM_Core_Controller_Simple(
      'CRM_Activity_Form_ActivityLinks',
      ts('Activity Links'),
      NULL,
      FALSE, FALSE, TRUE
    );
    $controller->setEmbedded(TRUE);
    $controller->run();
  }
```